Index: linyaps-box/src/linyaps_box/container.cpp
===================================================================
--- linyaps-box.orig/src/linyaps_box/container.cpp
+++ linyaps-box/src/linyaps_box/container.cpp
@@ -1382,6 +1382,14 @@ void configure_mounts(linyaps_box::conta
         return ss.str();
     }();
 
+    // NOTE: This node only exist in UOS kernel, it is used to identify child process
+    // running in linyaps container
+    auto elf_verify_cred = std::filesystem::path{ "/proc/self/attr/elfverify/cred" };
+    if (std::filesystem::exists(elf_verify_cred)) {
+        std::ofstream ofs(elf_verify_cred);
+        ofs << "1";
+    }
+
     execvpe(c_args.at(0),
             const_cast<char *const *>(c_args.data()), // NOLINT
             const_cast<char *const *>(c_env.data())); // NOLINT
